### ISSUE TITLE
Ex

### ISSUE THREAD

**Issue by NanciuDavid** [NanciuDavid/DataAnalysis_Red-Wine#1]
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import scipy.cluster.hierarchy as sch
from sklearn.preprocessing import StandardScaler

# Load data
df_alcohol = pd.read_csv("./dataIN/alcohol.csv", index_col=0)
df_countries = pd.read_csv("./dataIN/CoduriTariExtins.csv", index_col=0)

# Columns for years
years = ["2000", "2005", "2010", "2015", "2018"]

# Requirement 1: Calculate mean alcohol consumption per country
df_alcohol["MeanConsumption"] = df_alcohol[years].mean(axis=1)
df_alcohol[["Code", "MeanConsumption"]].to_csv("./dataOUT/cerinta1.csv")

# Requirement 2: Find the year with max continental consumption
df_merged = df_alcohol.merge(df_countries, left_index=True, right_index=True)
df_continent = df_merged.groupby("Continent")[years].mean()
max_year = df_continent.mean(axis=0).idxmax()
max_continent = df_continent[max_year].idxmax()
pd.DataFrame({"Continent_Name": [max_continent], "Anul": [max_year]}).to_csv("./dataOUT/cerinta2.csv")

# Requirement 3: Standardize alcohol consumption data
df_standardized = pd.DataFrame(
    StandardScaler().fit_transform(df_alcohol[years]),
    index=df_alcohol.index, columns=years
)

# Perform hierarchical clustering
linked = sch.linkage(df_standardized, method="ward")

# Save hierarchical clustering matrix to console
hierarchical_matrix = np.column_stack((linked[:, :2], linked[:, 2], np.arange(1, len(linked) + 1)))
print("Hierarchical Clustering Matrix:\n", hierarchical_matrix)

# Requirement 4: Plot dendrogram
plt.figure(figsize=(10, 6))
sch.dendrogram(linked, labels=df_alcohol.index, leaf_rotation=90, leaf_font_size=10)
plt.title("Dendrogram - Ward Clustering")
plt.xlabel("Countries")
plt.ylabel("Distance")
plt.axhline(y=linked[-5, 2], color="r", linestyle="--")  # Example: cutting at 5 clusters
plt.show()

# Requirement 5: Assign clusters and save
from scipy.cluster.hierarchy import fcluster
df_alcohol["Cluster"] = fcluster(linked, t=5, criterion="maxclust")
df_alcohol[["Code", "Cluster"]].to_csv("./dataOUT/popt.csv")
